src/components/organisms/shared/OakPagination/OakPagination.tsx

```diff
@@ -137,20 +140,53 @@ export const OakPagination = ({
   prevHref,
   paginationHref,
   pageName,
+  firstItemRef,
+  router,
+  shouldAppendQuery,
 }: OakPaginationProps) => {
+  useEffect(() => {
+    if (router.query.page && firstItemRef?.current) {
+      firstItemRef.current.focus();
+      window.scrollTo({ top: 0, behavior: "smooth" });
+    }
+  }, [firstItemRef, router.query.page]);
+
   const [activePage, setActivePage] = useState(currentPage);
```

Comment:
This is an existing bug but there is no way for the consumer to change `currentPage` after the first render which would be surprising for anyone using this component.

We could:

a. rename this to `initialPage` (this is a common convention for naming props that can only be set once)
b. ditch the internal `activePage` and use of `useState` and always respect the value of `currentPage` — that would leave it up to the consumer to maintain the state of the current page (which we can get from the router).

E.g.

```typescript
const currentPage = router.query.page ? Number(router.query.page) : 1;
<OakPagination currentPage={currentPage} />
```

Reply:
Renamed to `initialPage` ✅